apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -1210,9 +1193,21 @@ defmodule Champions.Battle.Simulator do
         :execution_received
       )
 
+    initial_interval_steps = get_interval_steps(execution_over_time.execution)
+
     new_target =
       target
       |> Map.put(:health, target.health - damage_after_defense)
+      |> update_in([:executions_over_time], fn current_executions ->
+        Enum.map(current_executions, fn exec ->
+          if exec.skill_id == execution_over_time.skill_id do
```

Comment:
Correct me if I'm wrong, but this will fail as the same execution over time could be applied to a unit twice (imagine a very short cooldown, or 2 same heroes in a fight. Also, two different executions that have the same `skill_id`). That would probably break this implementation: `remaining_duration` would decrease by `n` every step instead of 1, `n` being the amount of times `skill_id` is repeated among the target